contributed_definitions/nyaml/NXtransfer_matrix_table.yaml

```diff
@@ -0,0 +1,48 @@
+category: base 
+doc: |
+    Transfermatrix relationship between N input beams and M output beams.
+    It contains a table with the relevant matricis to be used for different
+    transmissitted properties (such as polarization, intensity, phase).
+
+    Data structure for all transfermatrices of an optical element in a setup.
+    For each combination of N input and M output beams and for L physical
+    concept (i.e. beam intensity), one matrix can be defined.
+
+    In this way, the transfermatrix table has the dimension NxM.
+
+    For each entry, in this transfermatrix, there are L formalisms.
+    Each formalism has the dimension math:`dim(L_i)xdim(L_i)`,
+    whereby math:`L_i` is the specific physical concept (Intensity, polarization, direction).
+
+    A beamsplitter with two input laser beams can have a total of
+    four transfermatrices (2 Input x 2 Output).
+
+    The dimension of the transfermatrix depends on the parameters.
+    Examples are:
+    1x1 for intensity/power
+    2x2 for jones formalism
+    3x3 for direction
+symbols:
+  N_in_out: |
+    Dimension of number total number of input and output combinations.
+    i.e. two input and two output --> dim 4
+  N_variables: |
+    Dimension of the number of colums for the data array, which stores
+    a scalar (dim1) or vector (dim3)
+type: group
+transfer_matrix_table(NX_NUMBER): # (NXdata)?

```

Comment:
NXtransfer_matrix_table

Again, this should extend NXobject, not NX_NUMBER.

Reply:
resturctured